### ISSUE TITLE
Offering assistance and discussing upkeep / releases for vtprotobuf

### ISSUE THREAD

**Issue by nockty** [planetscale/vtprotobuf#89]
Hello everyone,

I'd like to discuss the future maintenance of vtprotobuf. As an avid user from [Datadog](https://github.com/DataDog), I've recognized its potential in enhancing protobuf message operations. However, recent activity in the repository has been limited, with the last release dating back to January.

To bolster vtprotobuf's capabilities for developers, I propose exploring a more active maintenance and release cycle. Acknowledging the demands of open-source projects, I'm here to extend a hand, along with some of my coworkers from Datadog, to offer assistance.

Although notable bugs and promising ideas exist in pull requests and issues (like https://github.com/planetscale/vtprotobuf/pull/83 and https://github.com/planetscale/vtprotobuf/issues/54), no PRs have been merged since the beginning of the year. We're enthusiastic about bridging this gap and improving vtprotobuf's efficiency and robustness.

In this regard, I suggest opening a discussion on project maintenance and future releases. Our goal is collaborative growth, not imposition. Whether it's contributing code, managing issues, updating documentation, or handling releases, we are more than willing to step in.

Let's collectively ensure that vtprotobuf remains a valuable resource for developers. Your insights are vital, and I'm excited about the potential improvements we can achieve.

Looking forward to your thoughts and suggestions. Thank you for your time and consideration.

**Comment by vmg** [comment]
Hey @nockty, thanks for reaching out. You actually caught me off on paternity leave, which is the reason why the repository was basically frozen for the better part of this year. This is my bad -- I did not hand over this specific project before I left. Sorry about that!

But I'm back now, and really happy that DD is interested on helping maintain the library.

I know that lots of people rely on `vtprotobuf` at this point, including Vitess, but the project already does 100% of what we required it to do for Vitess RPC, so I've been finding it hard to spend more of my time working on it. Having maintainers that are not tied to Vitess itself will be helpful to ensure that `vtprotobuf` covers more use cases.

I'm gonna start by discussing the security implications of this internally. We may require to spin-off this into its own organization before we can add external contributors to the project. I'll let you know asap. Thanks again for your interest!

**Comment by gnagel** [comment]
Hi @vmg ,
I too would like to offer support from [TriggerMail](https://github.com/TriggerMail) to maintain this too!  This is an awesome library and I would love to see it supported going forward.
Thanks,
Glenn

**Comment by nockty** [comment]
Hi @vmg, thank you for getting back to us, and congratulations on your paternity leave! It's completely understandable that life sometimes takes precedence.

I'm glad to hear that you're back and the project is getting attention again. It's exciting that DD's interest in maintaining vtprotobuf can align with the project's future and it's wonderful to see such positive engagement from the TriggerMail team as well.

I'm really looking forward to hearing from you about the next steps. Thanks for your responsiveness and dedication to the project!

**Comment by biosvs** [comment]
Hi,

@nockty thanks for initiative and for rising this question!

@vmg here at Uber we're also looking for an alternative to outdated and abandoned gogo/protobuf project. And vtProto seems really promising!

But still we want to understands the desired future of this project:
1. Does (will?) project have multiple owners/admins who may support activity, including sorting out issues, PRs and new releases?
2. Who and how is (will?) making decision about features and approaches: what should be implemented, what should be left out of scope, etc.

As we're interested in this project, we also offer our help and participation in this project along with @nockty and @gnagel (at companies behind them) who already expressed their will.

Thanks!

**Comment by vmg** [comment]
As a hint for people interesting on contributing to the library: I think the main open issue right now blocking our next release is (de)serialization support for well-known types. There's an open PR that implements it, but it's large and complex and I haven't gotten around to reviewing it yet. Testing it in your own projects and giving feedback would be immensely helpful.

**Comment by vmg** [comment]
@nockty: :wave: welcome back from your holidays! I've given you triage access to the repository. Thanks again for your interest on the project and let me know what can I do to make your life easier.

**Comment by nockty** [comment]
Awesome, thank you for your trust @vmg. I'll do my best to ensure that `vtprotobuf` succeeds in covering more use cases. I'm also thrilled to see that the project is getting active again, that's exciting!

I'll let you know if I need anything but in the meantime feel free to provide feedback at anytime so that I'm helping the project move in the right direction.

**Comment by anu-tiernan** [comment]
speaking of, it looks like there's a fix for oneof encoding in main but there hasn't been a release with that fix yet. Is there anything blocking cutting a new release?
